lib/features/give/pages/bt_scan_page.dart

```diff
@@ -76,13 +103,31 @@ class _BTScanPageState extends State<BTScanPage> {
       if (!mounted) {
         return;
       }
+
+      // Check if its really a Givt beacon
+      final hex = StringBuffer();
+      for (final b in scanResult.advertisementData
+          .serviceData[scanResult.advertisementData.serviceUuids.first]!) {
+        hex.write(sprintf('%02x', [b]));
+      }
+
+      final beaconData = hex.toString();
+      final contains = beaconData.contains('61f7ed01') ||
+          beaconData.contains('61f7ed02') ||
+          beaconData.contains('61f7ed03');
+
+      if (!contains) {
+        return;
+      }
+
```

Comment:
Yeah ok, I guess it's better here. But ee could have created a new cubit with only the Bluetooth instance there and process the data and just pass it into the give_bloc, what do you think? 🤔